### PR TITLE
feat: display name from magnet link when metadata is missing

### LINKED ISSUE #117: Display torrent name when downloading metadata if available via magnet link
Magnet links can contain `dn` which is the display name of the torrent. We should show that instead of "unknown" when downloading such metadata

### PR DESCRIPTION
**Pull request by tkalum**
I've implemented #117 by adding support for showing the magnet link name (`dn` parameter) in the UI immediately, instead of displaying "unknown" until metadata downloads.

The changes include:

    Refactored parse_magnet_link to return a struct with info hash and optional name.

    Added a display_name field to VortexApp and updated the constructor.

    Passed the name from main to the app and used it in the UI render.

Closes #117

### CODE REVIEW

File: cli/src/app.rs [hunk 1]
```diff
@@ -67,9 +67,11 @@ pub struct VortexApp<'queue> {
     pub shutdown_signal_tx: Sender<()>,
     pub state: AppState,
     pub dht_paused: Arc<AtomicBool>,
+    pub magnet_name: Option<String>,
 }
 
 impl<'queue> VortexApp<'queue> {
+    #[allow(clippy::too_many_arguments)]
```
Comment (Nehliin) [inline cli/src/app.rs]: Instead of adding a new argument here I think we can make the existing `metadata` argument something like this:

```
enum Metadata {
     Full(Box<lava_torrent....>),
     Partial {
         name: String
     }.
     None
}
```

Then we do not need to add an additional field to the struct either

File: cli/src/main.rs [hunk 2]
```diff
@@ -300,6 +316,8 @@ fn main() -> color_eyre::Result<()> {
     let bt_config = vortex_config.bittorrent;
     let root = paths.download_folder.clone();
 
+    let mut magnet_name: Option<String> = None;
```
Comment (Nehliin) [inline cli/src/main.rs]: This can then be removed and you can instead return (mut state, metadata) as a tuple from the match statement handling `cli.torrent_info`

File: cli/src/main.rs [hunk 3]
```diff
@@ -478,6 +501,7 @@ impl Widget for &mut VortexApp<'_> {
                 .metadata
                 .as_ref()
                 .map(|meta| meta.name.to_owned())
+                .or_else(|| self.magnet_name.clone())
```
Comment (Nehliin) [inline cli/src/main.rs]: this then becomes a match statement for the new metadata enum

File: cli/src/main.rs [hunk 4]
```diff
@@ -516,4 +544,15 @@ mod tests {
             "magnet:?xt=urn:btmh:1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef";
         assert!(parse_magnet_link(link).is_err());
     }
+
+    #[test]
+    fn test_parse_magnet_without_name() {
```
Comment (Nehliin) [inline cli/src/main.rs:559]: nice, could we add one or two more tests checking for resilience when the magnet link is misformed like an empty display name for example?

**Review by Nehliin — CHANGES_REQUESTED** [review]
Nice start!

**Comment by tkalum** [comment]
I’ve completed the refactor for the metadata logic. Here is what I changed:

    Replaced Option<TorrentMetadata> with a custom Metadata enum.

    Updated the UI logic to handle partial metadata names correctly.

    Added unit tests for magnet link edge cases.

    Ensured an 'Unknown' fallback for missing or invalid display names.

About the CI error: I ran cargo update to fix some security warnings, but now the cargo deny check is failing because of 'duplicate' crates. I’m not sure how to fix these version conflicts in the deny.toml file, so I wanted to let you know. I also ran cargo fmt and clippy to keep the code clean!

**Comment by Nehliin** [comment]
I've fixed the deny issues in #120, you can rebase this pr on top of master and then we should be good to go :)

**Comment by tkalum** [comment]
Okay, should be good now.

**Review by Nehliin — APPROVED** [review]
thanks!
